evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -40,21 +40,23 @@ def __init__(self, boosting_type="gbdt", learning_rate=0.1, n_estimators=100, ma
                       "max_depth": max_depth,
                       "num_leaves": num_leaves,
                       "min_child_samples": min_child_samples,
-                      "n_jobs": n_jobs,
-                      "bagging_freq": bagging_freq,
-                      "bagging_fraction": bagging_fraction}
+                      "n_jobs": n_jobs}
         parameters.update(kwargs)
-
+        lg_parameters = copy.copy(parameters)
         # when boosting type is random forest (rf), LightGBM requires bagging_freq == 1 and  0 < bagging_fraction < 1.0
-        if boosting_type == "rf" and not bagging_freq:
-            parameters.update({'bagging_freq': 1})
+        if boosting_type == "rf":
+            # bagging_fraction could be passed in as a kwarg, which we wouldn't want to overwrite
```

Comment:
I think we should override bagging_fraction if it's equal to 1 or 0 or else we end up back at #1251 ?